From 57d95122bb959afe91017a9d601e513c33d9369f Mon Sep 17 00:00:00 2001
From: sakshi <122897122+s4k5sh1@users.noreply.github.com>
Date: Tue, 6 Aug 2024 12:30:23 +0100
Subject: [PATCH] commit chnages to client3Update client3.py to improve
 performance

python -u "/Users/sakshisingh/Desktop/JP-Morgan-Virtual-Internship---Software-Engineepython -u "/Users/sakshisingh/Desktop/JP-Morgan-Virtual-Internship---Software-Engineering-1/client3.py"
python -u "/Users/sakshisingh/Desktop/JP-Morgan-Virtual-Internship---Software-Engineering-1/client3.py"
python -u "/Users/sakshisingh/Desktop/JP-Morgan-Virtual-Internship---Software-Engineering-1/client3.py"
python -u "/Users/sakshisingh/Desktop/JP-Morgan-Virtual-Internship---Software-Engineering-1/client3.py"
python -u "/Users/sakshisingh/Desktop/JP-Morgan-Virtual-Internship---Software-Engineering-1/client3.py"
git commit -m "Your commit message"

modified:   client3.py
---
 client3.py | 48 ++++++++++++++++++++++++++----------------------
 1 file changed, 26 insertions(+), 22 deletions(-)

diff --git a/client3.py b/client3.py
index 3fc09b7..acd5578 100644
--- a/client3.py
+++ b/client3.py
@@ -18,9 +18,10 @@
 #  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 #  DEALINGS IN THE SOFTWARE.
 
+import urllib.request
+import time
 import json
 import random
-import urllib.request
 
 # Server API URLs
 QUERY = "http://localhost:8080/query?id={}"
@@ -28,32 +29,35 @@ QUERY = "http://localhost:8080/query?id={}"
 # 500 server request
 N = 500
 
-
 def getDataPoint(quote):
-    """ Produce all the needed values to generate a datapoint """
-    """ ------------- Update this function ------------- """
-    stock = quote['stock']
-    bid_price = float(quote['top_bid']['price'])
-    ask_price = float(quote['top_ask']['price'])
-    price = bid_price
-    return stock, bid_price, ask_price, price
-
+	""" Produce all of the needed values to generate a datapoint """
+	""" ------------- Update this function ------------- """
+	stock = quote['stock']
+	bid_price = float(quote['top_bid']['price'])
+	ask_price = float(quote['top_ask']['price'])
+	price = float((bid_price+ask_price)/2)
+	#print( (stock, bid_price, ask_price, price))
+	return (stock, bid_price, ask_price, price)
 
 def getRatio(price_a, price_b):
-    """ Get ratio of price_a and price_b """
-    """ ------------- Update this function ------------- """
-    return 1
-
+	""" Get ratio of price_a and price_b """
+	""" ------------- Update this function ------------- """
+	""" Also create some unit tests for this function in client_test.py """
+	if price_b == 0:
+		return
+	return float(price_a/price_b)
 
 # Main
 if __name__ == "__main__":
-    # Query the price once every N seconds.
-    for _ in iter(range(N)):
-        quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
-        """ ----------- Update to get the ratio --------------- """
-        for quote in quotes:
-            stock, bid_price, ask_price, price = getDataPoint(quote)
-            print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
+	# Query the price once every N seconds.
+	for _ in iter(range(N)):
+		quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
+
+		prices = {}
+		for quote in quotes:
+			stock, bid_price, ask_price, price = getDataPoint(quote)
+			prices[stock] = price
+			print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-        print("Ratio %s" % getRatio(price, price))
+		print ("Ratio %s" % getRatio(prices["ABC"], prices["DEF"]))
\ No newline at end of file
-- 
2.39.3 (Apple Git-146)

